File: week2/day3/diagonal-matrix.py

# - Create (dynamically) a two dimensional list
#   with the following matrix. Use a loop!
#
#   1 0 0 0
#   0 1 0 0
#   0 0 1 0
#   0 0 0 1
#
# - Print this two dimensional list to the output

a = []
for x in range(4):
    row = []
    for y in range(4):
        row.append(0)
    a.append(row)


a[0][0] = 1
a[1][1] = 1
a[2][2] = 1
a[3][3] = 1

print (a[0][0], a[0][1], a[0][2], a[0][3])
print (a[1][0], a[1][1], a[1][2], a[1][3])
print (a[2][0], a[2][1], a[2][2], a[2][3])
print (a[3][0], a[3][1], a[3][2], a[3][3])

File: week8/day3/diagonal-matrix.js
'use strict';
// - Create (dynamically*) a two dimensional list
//   with the following matrix**. Use a loop!
//
//   0 0 0 1
//   0 0 1 0
//   0 1 0 0
//   1 0 0 0
//
// - Print this two dimensional list to the console
//
// * size should depend on a variable
// ** Relax, a matrix is just like an array


function matrixer(a) {
    var matrix = [];
    for (var j = 0; j < a; j++) {
        matrix[j] = [];
        for (var i = 0; i < a; i++) {
            matrix[j][i] = 0;
        }
    }
    for (var k = 0; k < a; k++) {
        matrix[k][(a-1-k)] = 1;
    }
    console.log(matrix);
}

matrixer(10);
